src/lib/comms/include/sol-http-server.h

```diff
@@ -56,25 +56,126 @@ extern "C" {
+
+/**
+ * Deletes the server and consequently all handlers and dir added.
+ *
+ * @param server The value got with @c sol_http_server_new
+ */
 void sol_http_server_del(struct sol_http_server *server);
+
+/**
+ * Register a handler for a specific path.
+ *
+ * @param server The value got with @c sol_http_server_new
+ * @param path The path where the handler will serve, it means, when a request comes in this path the
+ * callback set will be called and a response should be sent back
+ * @param request_cb The callback for the request received on the specified path
+ * @param data The user data passed in the callback
+ *
+ * @return '0' on success, error code (always negative) otherwise.
+ */
 int sol_http_server_register_handler(struct sol_http_server *server, const char *path,
     int (*request_cb)(void *data, struct sol_http_request *request),
     const void *data);
+
+/**
+ * Removes a handler registered with @c sol_http_server_register_handler
+ *
+ * @param server The value got with @c sol_http_server_new
+ * @param path The same path given on @c sol_http_server_register_handler
+ *
+ * @return '0' on success, error code (always negative) otherwise.
+ */
 int sol_http_server_unregister_handler(struct sol_http_server *server, const char *path);
 
-/* The http server will look first for a handler when a request comes,
+/**
+ * Add a root dir where the server will look for static files to serve.
+ *
+ * @note: The http server will look first for a handler when a request comes,
  * if no valid handler is found it will try to find the file in the
  * root dirs set. The response will be sent as soon as a file matches
  * with the request.
+ *
+ * @param server The value got with @c sol_http_server_new
+ * @param rootdir The dir where the server will look for static files
+ *
+ * @return '0' on success, error code (always negative) otherwise.
  */
 int sol_http_server_add_dir(struct sol_http_server *server, const char *rootdir);
+
+/**
+ * Removes a dir registered with @c sol_http_server_add_dir
+ *
+ * @param server The value got with @c sol_http_server_new
+ * @param rootdir The same rootdir given on @c sol_http_server_add_dir
+ *
+ * @return '0' on success, error code (always negative) otherwise.
+ */
 int sol_http_server_remove_dir(struct sol_http_server *server, const char *rootdir);
+
+/**
+ * Set the last time the specified path had its value modified. It'll make the server
+ * return automatically a response with the code 304 (not modified) when the request contains the header
+ * If-Since-Modified greater than the value given in this function.
+ *
+ * @param server The value got with @c sol_http_server_new
+ * @param path The same path given on @c sol_http_server_register_handler
+ * @param modified The time it was modified
+ *
+ * @return '0' on success, error code (always negative) otherwise.
+ */
 int sol_http_server_set_last_modified(struct sol_http_server *server, const char *path, time_t modified);
```

Comment:
are we able to server something that is not a file?

if we're generating index.html, then we can use the newest file as the index.html mtime.